persistence/src/main/java/edu/unc/lib/dl/persist/services/edit/EditFilenameService.java

```diff
@@ -67,17 +68,21 @@ public void editLabel(AgentPrincipals agent, PID pid, String label) {
 
         try (Timer.Context context = timer.time()) {
             aclService.assertHasAccess(
-                    "User does not have permissions to edit labels",
+                    "User does not have permissions to edit filenames",
                     pid, agent.getPrincipals(), Permission.editDescription);
 
             RepositoryObject obj = repoObjLoader.getRepositoryObject(pid);
 
+            if (!(obj instanceof FileObject)) {
+                throw new IllegalArgumentException("Failed to edit filename for " + obj.getPid());
+            }
+
             String oldLabel = getOldLabel(obj);
 
-            repoObjFactory.createExclusiveRelationship(obj, DcElements.title, label);
+            repoObjFactory.createExclusiveRelationship(obj, Ebucore.filename, label);
```

Comment:
actually I'm realizing that that this relationship needs to be set on the binary rather than the FileObject. You can get the binary by calling `fileObj.getOriginalFile()`. You'll also need to get the old label from the same place. Sorry I didn't notice that sooner, I haven't gotten to manually test yet.